Assets/MixedRealityToolkit.Providers/WindowsMixedReality/WindowsMixedRealityDeviceManager.cs

```diff
@@ -390,12 +390,24 @@ private void GetOrAddController(InteractionSourceState interactionSourceState)
 
             if (controller != null)
             {
-                if (raiseSourceDetected)
+                var mrtkController = controller as WindowsMixedRealityController;
+
+                if (mrtkController != null)
+                {
+                    await mrtkController.EnsureControllerModel(interactionSourceState.source);
+                }
+
+                // Does the controller still exist after we loaded the controller model?
```

Comment:
If not, should we clean anything up? The model?

Reply:
I remember a mechanism that was caching the model so we do not have to load it again every time you come back from the shell. Since the primary goal is usually to keep both motion controllers active and it's just 2 of them, wouldn't keeping the model make sense?

Reply:
Yeah, that's true! As long as the model correctly disables itself after load. We can run some tests for this case post-merge